buffer/src/main/java/io/netty/buffer/PooledByteBufAllocator.java

```diff
@@ -233,41 +324,297 @@ protected ByteBuf newDirectBuffer(int initialCapacity, int maxCapacity) {
-            }
+            buf = PlatformDependent.hasUnsafe() ?
+                    UnsafeByteBufUtil.newUnsafeDirectByteBuf(this, initialCapacity, maxCapacity) :
+                    new UnpooledDirectByteBuf(this, initialCapacity, maxCapacity);
         }
 
         return toLeakAwareBuffer(buf);
     }
 
+    /**
+     * Default number of heap areanas - System Property: io.netty.allocator.numHeapArenas - default 2 * cores
+     */
+    public static int defaultNumHeapArena() {
+        return DEFAULT_NUM_HEAP_ARENA;
+    }
+
+    /**
+     * Default numer of direct arenas - System Property: io.netty.allocator.numDirectArenas - default 2 * cores
+     */
+    public static int defaultNumDirectArena() {
+        return DEFAULT_NUM_DIRECT_ARENA;
+    }
+
+    /**
+     * Default buffer page size - System Property: io.netty.allocator.pageSize - default 8192
+     */
+    public static int defaultPageSize() {
+        return DEFAULT_PAGE_SIZE;
+    }
+
+    /**
+     * Default maximum order - System Property: io.netty.allocator.maxOrder - default 11
+     */
+    public static int defaultMaxOrder() {
+        return DEFAULT_MAX_ORDER;
+    }
+
+    /**
+     * Default tiny cache size - System Property: io.netty.allocator.tinyCacheSize - default 512
+     */
+    public static int defaultTinyCacheSize() {
+        return DEFAULT_TINY_CACHE_SIZE;
+    }
+
+    /**
+     * Default small cache size - System Property: io.netty.allocator.smallCacheSize - default 256
+     */
+    public static int defaultSmallCacheSize() {
+        return DEFAULT_SMALL_CACHE_SIZE;
+    }
+
+    /**
+     * Default normal cache size - System Property: io.netty.allocator.normalCacheSize - default 64
+     */
+    public static int defaultNormalCacheSize() {
+        return DEFAULT_NORMAL_CACHE_SIZE;
+    }
+
+    /**
+     * Return {@code true} if direct memory cache aligment is supported, {@code false} otherwise.
+     */
+    public static boolean isDirectMemoryCacheAlignmentSupported() {
+        return PlatformDependent.hasUnsafe();
+    }
+
     @Override
     public boolean isDirectBufferPooled() {
         return directArenas != null;
     }
 
-//    Too noisy at the moment.
-//
-//    public String toString() {
-//        StringBuilder buf = new StringBuilder();
-//        buf.append(heapArenas.length);
-//        buf.append(" heap arena(s):");
-//        buf.append(StringUtil.NEWLINE);
-//        for (PoolArena<byte[]> a: heapArenas) {
-//            buf.append(a);
-//        }
-//        buf.append(directArenas.length);
-//        buf.append(" direct arena(s):");
-//        buf.append(StringUtil.NEWLINE);
-//        for (PoolArena<ByteBuffer> a: directArenas) {
-//            buf.append(a);
-//        }
-//        return buf.toString();
-//    }
+    /**
+     * Returns {@code true} if the calling {@link Thread} has a {@link ThreadLocal} cache for the allocated
+     * buffers.
+     */
+    @Deprecated
+    public boolean hasThreadLocalCache() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)